src/treemodel/delta/delta.js

```diff
@@ -118,6 +131,31 @@ export default class Delta {
 	}
 
 	/**
+	 * Custom toJSON method to make deltas serializable.
+	 *
+	 * @returns {Object} Clone of this delta with added class name.
+	 */
+	toJSON() {
+		let json = clone( this );
+
+		json.__class = this.constructor.className;
```

Comment:
I've used `__class` to be shorter but this is not so critical so can be `__className` also.